### PR TITLE
Update animation for onboarding carousel.

### LINKED ISSUE #15992: Add swiping - user should be able to swipe between the slideshow and navigate

### LINKED ISSUE #16221: Add swiping animation to carousel.
# Add swiping animation to carousel.

Add animation while drag is happing while swiping the onboarding carousel.

### PR DESCRIPTION
**Pull request by rahulpsingh**
fixes #16221 

### Summary

- Update progress on swiping carousel.

### Recording
recording

### Testing notes
- Long press on the carousel should stop the progress.
- Swiping on the carousel should swipe with animation.

#### Platforms

- Android
- iOS

##### Functional

- Onboarding

### Steps to test

- Open Status
- Try and navigate on the Onboarding carousel.

status: ready

### CODE REVIEW

**Review by ajayesivan — COMMENTED** [review]
Code LGTM!

@rahulpsingh I have a suggestion for improvement. Currently, when I release my hand in the middle of a swipe, the image instantly jumps into position. It would be great if we could add an animation for a smoother transition in such cases as well.

Also, I can swipe infinitely to the left, but not to the right.


https://github.com/status-im/status-mobile/assets/19279756/f566993b-88d0-4491-917d-f4f1dbdddad1

**Comment by rahulpsingh** [comment]
@ajayesivan Sure, yes!! I'll ask @xAlisher for a recommendation about swiping animation and implement it.

> Also, I can swipe infinitely to the left, but not to the right.

This is an intended feature.

**Comment by J-Son89** [comment]
@rahulpsingh please consider updating the animation in a follow up. Pr looks good to go 👌

File: src/status_im2/contexts/onboarding/common/carousel/animation.cljs
Comment (J-Son89) [inline src/status_im2/contexts/onboarding/common/carousel/animation.cljs:88]: 👌

**Review by J-Son89 — APPROVED** [review]
nice work @rahulpsingh!

**Comment by pavloburykh** [comment]
@rahulpsingh thanx for the PR. Please, take a look at the following issues:

### ISSUE 1 Delayed Slide Response in Carousel Swipe Interaction

**Steps:** 

1. Interact with onboarding slides
2. Pay attention at response for interactions

**Actual result:** 

When interacting with the carousel by swiping the slides, there is a noticeable delay between the user's swipe gesture and the actual transition of the slide. This delay causes a poor response, as there is a significant lag between the user's movement and the slide switch. It appears that the slide transition is not happening immediately upon the user's swipe action, resulting in a delayed and unresponsive experience. 


https://github.com/status-im/status-mobile/assets/97245802/5b944a89-ce4f-4068-b93c-6d388f816da9

**Comment by pavloburykh** [comment]
### ISSUE 2 Abrupt Slide Transition without Animation during Swipe Interaction

**Steps:** 

1. Swipe the slides
2. Pay attention at transition between slides during swiping

**Actual result:** 

During the swipe interaction of the slides, the transition between them is abrupt. This means that one slide abruptly changes to the next one without any visible transition or smooth animation between them. The moment of transition between slides is not perceptible, causing a jarring and sudden switch from one slide to another. This lack of smoothness in the slide animation negatively affects the overall user experience, as it hinders the ability to perceive and appreciate the visual flow and continuity of the content.


https://github.com/status-im/status-mobile/assets/97245802/b84651b0-232e-42fd-b07b-3ff1638b5d52

File: src/js/worklets/onboarding_carousel.js
Comment (ajayesivan) [inline src/js/worklets/onboarding_carousel.js]: If page size is `100/totalPages` then imo it would be better explicitly put that in the code.

@rahulpsingh wdyt?

**Review by ajayesivan — APPROVED** [review]
Nice work @rahulpsingh 🎉

**Comment by rahulpsingh** [comment]
@pavloburykh Both of the issues has been addressed. Please just have a look once!!

**Comment by pavloburykh** [comment]
@rahulpsingh thanx for the fixes!

[ISSUE 2](https://github.com/status-im/status-mobile/pull/16294#issuecomment-1600772452) - fixed

[ISSUE 1](https://github.com/status-im/status-mobile/pull/16294#issuecomment-1600772322) - IMO response is still bit delayed. But I would not block this PR by this issue. We will ask @status-im/design-team to try it out after merging and will log a followup if needed. 

Please also take a look at ISSUE 3. We can log it as a followup if this is not an easy fix, WDYT @rahulpsingh ?

### ISSUE 3  Long press does not stop the progress of the carousel

**Actual result:** in order to stop the progress of carousel user needs to at least slightly move the slide and then longpress. Pressing without moving does not result in stopping the progress. 


https://github.com/status-im/status-mobile/assets/97245802/308d131e-601c-4301-874b-7c70dce20f32

**Comment by pavloburykh** [comment]
@rahulpsingh here is a blocker that needs to be fixed

### ISSUE 4 "No protocol IDeref. -deref defined for type null" crash on generating keys

**Steps:** 

1. Proceed with creating new user
2. Observe crash after Generating keys step

**Actual result:**

photo_2023-06-27 11 09 28

https://github.com/status-im/status-mobile/assets/97245802/00fce652-44a3-4c43-b63f-efb6d627a087

**Comment by rahulpsingh** [comment]
@pavloburykh I'm looking at the crash issue(I think it's a side effect after rebase). Yes, I agree to the point that We can merge this PR and move to the design team and see their feedbacks as well.

Thank you!!

**Comment by rahulpsingh** [comment]
@pavloburykh The 4th issue should as well be fixed. Please just have a look once!!

**Comment by pavloburykh** [comment]
@rahulpsingh thanx for the fix. Ready for merge.

**Pull request merged**
